nengo/connection.py

```diff
@@ -122,20 +123,28 @@ class ConnectionSolverParam(SolverParam):
     def coerce(self, conn, solver):
         solver = super().coerce(conn, solver)
         if solver is not None:
-            if solver.weights and not isinstance(conn.pre_obj, Ensemble):
+            if solver.weights and not isinstance(conn.pre_obj, (Ensemble, Node)):
                 raise ValidationError(
                     "weight solvers only work for connections from ensembles "
                     "(got %r)" % type(conn.pre_obj).__name__,
                     attr=self.name,
                     obj=conn,
                 )
-            if solver.weights and not isinstance(conn.post_obj, Ensemble):
+            elif solver.weights and isinstance(conn.pre_obj, Node):
+                warnings.warn(
+                    "weight solvers only work for connections to ensembles, not nodes"
+                )
+            if solver.weights and not isinstance(conn.post_obj, (Ensemble, Node)):
                 raise ValidationError(
                     "weight solvers only work for connections to ensembles "
                     "(got %r)" % type(conn.post_obj).__name__,
                     attr=self.name,
                     obj=conn,
                 )
+            elif solver.weights and isinstance(conn.post_obj, Node):
+                warnings.warn(
+                    "weight solvers only work for connections to ensembles, not nodes"
```

Comment:
I wonder if we want to say a bit more about what's happening in this case. In this case, there's no difference between a solver with `weights=True` and `weights=False`, so maybe we want to make that clear (saying something like "setting `weights=True` on a solver has no effect when the post object is a Node"). Furthermore, the current statement is a bit misleading, because weight solvers do work, they just act the same as non-weight solvers.